Sekoia.io/sekoiaio/triggers/cases.py

```diff
@@ -0,0 +1,269 @@
+                and case_attrs.get("short_id") not in case_uuids_filter
+            ):
+                return False
+
+        return True
+
+    @retry(
+        reraise=True,
+        wait=wait_exponential(max=10),
+        stop=stop_after_attempt(10),
+    )
+    def _retrieve_case_from_caseapi(self, case_uuid):
+        api_url = urljoin(self.module.configuration["base_url"], f"api/v1/sic/cases/{case_uuid}")
+        api_url = api_url.replace("/api/api", "/api")  # In case base_url ends with /api
+
+        api_key = self.module.configuration["api_key"]
+        headers = {"Authorization": f"Bearer {api_key}", "User-Agent": user_agent()}
+
+        response = requests.get(
+            api_url,
+            headers=headers,
+            params={
+                "stix": False,
+                "comments": False,
+                "countermeasures": False,
+                "history": False,
+            },
+        )
+
+        if not response.ok:
+            try:
+                content = response.json()
+            except Exception:
+                content = response.text
+            self.log(
+                "Error while fetching case from Case API",
+                level="error",
+                status_code=response.status_code,
+                content=content,
+            )
+
+        # raise an exception if the http request failed
+        response.raise_for_status()
+        try:
+            return response.json()
+        except Exception as exp:
+            self.log("Failed to parse JSON response from Case API", level="error", content=response.text)
+            raise exp
+
+
+class CaseCreatedTrigger(SecurityCasesTrigger):
+    HANDLED_EVENT_SUB_TYPES = [("case", "created")]
+
+    def handle_event(self, message):
+        """Handle case created messages with filters."""
+        case_attrs = message.get("attributes", {})
+        event_type: str = message.get("type", "")
+        event_action: str = message.get("action", "")
+
+        # Ignore cases “sub event” types that we can’t (yet) handle.
+        if (event_type, event_action) not in self.HANDLED_EVENT_SUB_TYPES:
+            return
+
+        # Is the notification in a format we can understand?
+        case_uuid: str = case_attrs.get("uuid", "")
+        if not case_uuid:
+            return
+
+        if not self._filter_notifications(message):
+            return
+
+        try:
+            case = self._retrieve_case_from_caseapi(case_uuid)
+        except Exception as exp:
+            self.log_exception(exp, message="Failed to fetch case from case API")
+            return
+
+        work_dir = self._data_path.joinpath("sekoiaio_securitycases").joinpath(str(uuid.uuid4()))
+        case_path = work_dir.joinpath("case.json")
+        work_dir.mkdir(parents=True, exist_ok=True)
+
+        with case_path.open("w") as fp:
+            fp.write(orjson.dumps(case).decode("utf-8"))
+
+        directory = str(work_dir.relative_to(self._data_path))
```

Comment:
As all case's information fit in the returned result, IMHO, this part is not required (Indeed, unlike the alert triggers, the written file doesn't seem to be used in the code below).
```suggestion
```